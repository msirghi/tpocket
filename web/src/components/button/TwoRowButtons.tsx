import React from 'react';
import Button from "@material-ui/core/Button";
import { Row } from "../layout/Row";

interface IProps {
  buttonWidth: number;
  leftIcon: React.ReactNode;
  rightIcon: React.ReactNode;
  leftLabel: string;
  rightLabel: string;
  leftClickHandler: Function;
  rightClickHandler: Function;
}

export const TwoRowButtons: React.FC<IProps> = ({ buttonWidth, leftIcon, rightIcon, leftClickHandler, leftLabel, rightLabel, rightClickHandler }) => {
  return (
    <Row className={ 'row-center pb-15' }>
      <Button
        data-test={'left-button'}
        style={ { width: buttonWidth } }
        onClick={ () => leftClickHandler() }
        startIcon={ leftIcon }
        color="primary"
      >
        { leftLabel }
      </Button>
      <Button
        data-test={'right-button'}
        style={ { width: buttonWidth } }
        onClick={ () => rightClickHandler() }
        startIcon={ rightIcon }
        color="primary"
      >
        { rightLabel }
      </Button>
    </Row>
  );
}
