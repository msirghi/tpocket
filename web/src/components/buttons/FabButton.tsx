import React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Fab } from "@material-ui/core";

interface IProps {
  color: "inherit" | "default" | "primary" | "secondary" | undefined;
  label: string;
  onClick: Function;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      backgroundColor: theme.palette.background.paper,
      width: 500,
      position: 'relative',
      minHeight: 200,
    },
    fab: {
      position: 'fixed',
      bottom: theme.spacing(2),
      right: theme.spacing(2),
    },
  }),
);

export const FabButton: React.FC<IProps> = ({ color, label, children, onClick }) => {
  const classes = useStyles();

  return (
    <Fab data-test={'fab-button'} aria-label={ label } className={ classes.fab } color={ color } onClick={ () => onClick() }>
      { children! }
    </Fab>
  );
}
