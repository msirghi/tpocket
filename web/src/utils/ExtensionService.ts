const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export default {
  getMonthName
};
