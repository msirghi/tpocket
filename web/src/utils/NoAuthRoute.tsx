import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { getAccessToken } from '../accessToken';

// @ts-ignore
export const NoAuthRoute = ({ Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (getAccessToken() ? <Redirect to='/home' /> : <Component {...props} />)}
  />
);
