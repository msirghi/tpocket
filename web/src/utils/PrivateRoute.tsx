import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { getAccessToken } from '../accessToken';

// @ts-ignore
export const PrivateRoute = ({ Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (getAccessToken() ? <Component {...props} /> : <Redirect to='/login' />)}
  />
);
