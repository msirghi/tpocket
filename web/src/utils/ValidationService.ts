const passwordStrength = require('check-password-strength');

export const validateEmailStr = (email: string) => {
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
};

export const isPasswordWeak = (password: string): boolean => {
  return passwordStrength(password).value === 'Weak';
};

const validateName = (name: string) => {
  const re = /^(?=.{2,50}$)[a-z]+(?:['_.\s][a-z]+)*$/i;
  return re.test(String(name).toLowerCase());
};

export default {
  isPasswordWeak,
  validateEmailStr,
  validateName
};
