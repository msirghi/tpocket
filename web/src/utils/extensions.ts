const passwordStrength = require('check-password-strength');

const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export const validateEmailStr = (email: string) => {
  return !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
};

export const isPasswordWeak = (password: string): boolean => {
  return passwordStrength(password).value === 'Weak';
};
