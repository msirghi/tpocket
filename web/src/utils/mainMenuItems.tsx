import React from "react";
import { IMenuItem } from "../interfaces/IMenuItem";
import HomeIcon from '@material-ui/icons/Home';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import CategoryIcon from '@material-ui/icons/Category';
import HelpOutlineIcon from '@material-ui/icons/Help';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import EmailIcon from '@material-ui/icons/Email';

export const mainMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/home',
    name: 'Home',
    icon: <HomeIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/account',
    name: 'Account',
    icon: <AccountCircleIcon className={ 'light-grey-icon ' }/>
  },
  {
    id: 3,
    link: '/categories',
    name: 'Categories',
    icon: <CategoryIcon className={ 'light-grey-icon' }/>
  }
];

export const secondaryMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/help',
    name: 'Help',
    icon: <HelpOutlineIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/writeus',
    name: 'Support',
    icon: <EmailIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 3,
    link: '/logout',
    name: 'Sign out',
    icon: <ExitToAppIcon className={ 'light-grey-icon ' }/>
  }
];
