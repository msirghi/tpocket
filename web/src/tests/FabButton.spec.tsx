import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import { FabButton } from "../components/button/FabButton";
import { findByTestAttr } from "./utils/findByTestAttr";
import toJson from 'enzyme-to-json';

describe('FabButton component', () => {
  let component: ShallowWrapper;
  const onClick = jest.fn();

  beforeEach(() => {
    component = shallow(<FabButton color={'primary'} label={'Fab button'} onClick={onClick} />);
  });

  it('should render fab button', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should fire prop method on click', () => {
    const button = findByTestAttr(component, 'fab-button');
    button.simulate('click');
    expect(onClick).toBeCalledTimes(1);
  });
});
