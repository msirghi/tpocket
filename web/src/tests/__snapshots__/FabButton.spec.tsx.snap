// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabButton component should render fab button 1`] = `
<WithStyles(ForwardRef(Fab))
  aria-label="Fab button"
  className="makeStyles-fab-2"
  color="primary"
  data-test="fab-button"
  onClick={[Function]}
/>
`;
